File: app/src/main/java/za/co/woolworths/financial/services/android/models/dto/AddToListRequest.java
package za.co.woolworths.financial.services.android.models.dto;

public class AddToListRequest {

    private String skuID;
    private String giftListId;
    private String catalogRefId;
    private String quantity;
    private String listId;
    private String size;

    public String getSize() {
        return size;
    }

    public void setSize(String size) {
        this.size = size;
    }

    public void setSkuID(String skuID) {
        this.skuID = skuID;
    }

    public void setGiftListId(String giftListId) {
        this.giftListId = giftListId;
    }

    public void setCatalogRefId(String catalogRefId) {
        this.catalogRefId = catalogRefId;
    }

    public void setQuantity(String quantity) {
        this.quantity = quantity;
    }

    public String getSkuID() {
        return skuID;
    }

    public String getCatalogRefId() {
        return catalogRefId;
    }

    public String getGiftListId() {
        return giftListId;
    }

    public String getQuantity() {
        return quantity;
    }

    public void setListId(String listId) {
        this.listId = listId;
    }

    public String getListId() {
        return listId;
    }
}

File: app/src/main/java/za/co/woolworths/financial/services/android/models/dto/AddToListRequest.kt
package za.co.woolworths.financial.services.android.models.dto

import android.os.Parcelable
import kotlinx.parcelize.Parcelize
import za.co.woolworths.financial.services.android.contracts.FirebaseManagerAnalyticsProperties
import za.co.woolworths.financial.services.android.shoppinglist.service.network.CopyItemDetail
import za.co.woolworths.financial.services.android.util.analytics.dto.AnalyticProductItem

@Parcelize
data class AddToListRequest(
    var skuID: String? = null,
    var giftListId: String? = null,
    var catalogRefId: String? = null,
    var quantity: String? = null,
    var listId: String? = null,
    var isGWP: Boolean = false,
    var size: String? = null
) : Parcelable

fun AddToListRequest.toCopyItemDetail(): CopyItemDetail = CopyItemDetail(
    skuID = skuID ?: "",
    catalogRefId = catalogRefId ?: "",
    quantity = "1"
)

fun AddToListRequest.toAnalyticItem(category: String?) = AnalyticProductItem(
    itemId = skuID,
    category = category,
    itemListName = category,
    quantity = 1,
    affiliation = FirebaseManagerAnalyticsProperties.PropertyValues.AFFILIATION_VALUE,
    index = FirebaseManagerAnalyticsProperties.PropertyValues.INDEX_VALUE.toInt()
)
